services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/grpc/LogicalPortSpec.groovy

```diff
@@ -0,0 +1,78 @@
+package org.openkilda.functionaltests.spec.grpc
+
+
+import static org.junit.Assert.assertTrue
+import static org.openkilda.testing.ConstantsGrpc.LOGICAL_PORT
+import static org.openkilda.testing.ConstantsGrpc.SWITCH
+import static org.openkilda.testing.ConstantsGrpc.SWITCH_PORT
+
+import org.openkilda.functionaltests.BaseSpecification
+import org.openkilda.grpc.speaker.model.LogicalPortDto
+import org.openkilda.messaging.error.MessageError
+
+import groovy.util.logging.Slf4j
+import org.springframework.http.HttpStatus
+import org.springframework.web.client.HttpClientErrorException
+import spock.lang.Narrative
+import spock.lang.Unroll
+
+@Slf4j
+@Narrative("Test Logical Port.")
+class LogicalPortSpec extends BaseSpecification {
+    @Unroll
+    def "Not able to create logical port with incorrect ports number(lPort/sPort): #logicalPortNumber/#portNumber."() {
+        when:
+        " Try to create with incorrect values (logical port/simple port): $logicalPortNumber/$portNumber."
+        grpc.createLogicalPort(SWITCH, new LogicalPortDto([portNumber], logicalPortNumber))
+
+        then: "Human readable error is returned."
+        def exc = thrown(HttpClientErrorException)
+        assert exc.statusCode == HttpStatus.BAD_REQUEST
+        assert exc.responseBodyAsString.to(MessageError).errorMessage == errorMessage
+
+        where:
+        logicalPortNumber | portNumber  | errorMessage
+        99                | SWITCH_PORT | "Valid logicalportno range is 100 to 63487."
+        63488             | SWITCH_PORT | "Valid logicalportno range is 100 to 63487."
+        LOGICAL_PORT      | 44444       | "Invalid portno value."
+    }
+
+    def "Able to create/read/delete logicalport."() {
+        given: "Create logical port."
+        grpc.createLogicalPort(SWITCH, new LogicalPortDto([SWITCH_PORT], LOGICAL_PORT))
+
+
+        and: "Able to get the created logical port."
```

Comment:
This is not a `given`. Your test verifies create/read/update operations, so `Able to get the created logical port` cannot be 'given', because it is the behavior you are about to verify in the test. Same for the `given: "Create logical port."`